geode-assembly/src/acceptanceTest/java/org/apache/geode/jdk/JdkEncapsulationTest.java

```diff
@@ -0,0 +1,145 @@
+    int[] availablePorts = getRandomAvailableTCPPorts(2);
+    int locatorPort = availablePorts[0];
+    serverPort = availablePorts[1];
+    locatorString = "localhost[" + locatorPort + "]";
+    connectToLocator = "connect --locator=" + locatorString;
+
+    String startLocator = String.join(" ",
+        "start",
+        "locator",
+        "--port=" + locatorPort,
+        "--http-service-port=0");
+
+    Path jarPath = folderRule.getFolder().toPath().resolve("traverse-object.jar");
+    createJarWithClasses(jarPath, TraverseInaccessibleJdkObject.class);
+    String deployObjectTraverserFunction = String.join(" ",
+        "deploy",
+        "--jar=" + jarPath);
+    gfshRule.execute(startLocator, deployObjectTraverserFunction);
+  }
+
+  // If this test fails, it means the object we're trying to traverse has no inaccessible fields,
+  // and so is not useful for the other tests. If it fails, update TraverseInaccessibleJdkObject
+  // to use a type that actually has inaccessible fields.
+  @Test
+  public void cannotMakeInaccessibleFieldsAccessibleByDefault() {
+    gfshRule.execute(startServerWithOptions()); // No options
+
+    GfshExecution execution = GfshScript.of(connectToLocator)
+        .and(TRAVERSE_INACCESSIBLE_OBJECT)
+        .expectExitCode(1) // Because java.math is not opened by default.
+        .execute(gfshRule);
+
+    assertThat(execution.getOutputText())
+        .as("result of traversing %s", TraverseInaccessibleJdkObject.OBJECT.getClass())
+        .contains("Exception: java.lang.reflect.InaccessibleObjectException");
+  }
+
+  @Test
+  public void canMakeFieldsAccessibleOnExplicitlyOpenedPackages() {
+    String opensOptionFormat = "--J=--add-opens=%s/%s=ALL-UNNAMED";
+    String objectPackage = TraverseInaccessibleJdkObject.OBJECT.getClass().getPackage().getName();
+    String objectModule = TraverseInaccessibleJdkObject.MODULE;
+    String opensOption = String.format(opensOptionFormat, objectModule, objectPackage);
+    gfshRule.execute(startServerWithOptions(opensOption));
+
+    GfshScript.of(connectToLocator)
+        .and(TRAVERSE_INACCESSIBLE_OBJECT)
+        .expectExitCode(0) // Because we explicitly opened java.math.
+        .execute(gfshRule);
+  }
+
+  @Test
+  public void canMakeFieldsAccessibleOnPackagesOpenedByArgumentFile() {
+    // A few of the packages opened by this argument file are specific to Linux JDKs. Running this
+    // test on other operating systems might emit some warnings, but they are harmless.
+    gfshRule.execute(startServerWithOptions("--J=@" + LINUX_ARGUMENT_FILE_RELATIVE_PATH));
+
+    GfshScript.of(connectToLocator)
+        .and(TRAVERSE_INACCESSIBLE_OBJECT)
+        .expectExitCode(0) // Because the argument file opens java.math.
+        .execute(gfshRule);
+  }
+
+  private String startServerWithOptions(String... option) {
+    Stream<String> baseCommand = Stream.of(
+        "start",
+        "server",
+        "--name=server",
+        "--server-port=" + serverPort,
```

Comment:
You don't need a server-port if there are no clients. To disable it, you can use `--disable-default-server` and that just disables the server endpoint for handling client operations.

Reply:
Fixed.